src/main/groovy/util/StockMovementContext.groovy

```diff
@@ -0,0 +1,47 @@
+package util
+
+import org.pih.warehouse.api.StockMovementDirection
+import org.pih.warehouse.api.StockMovementType
+import org.pih.warehouse.auth.AuthService
+import org.pih.warehouse.core.Location
+import org.pih.warehouse.order.Order
+import org.pih.warehouse.requisition.Requisition
+import org.pih.warehouse.requisition.RequisitionSourceType
+import org.pih.warehouse.requisition.RequisitionStatus
+import org.pih.warehouse.shipping.Shipment
+
+class StockMovementContext {
+    Order order
+
+    Shipment shipment
+
+    Requisition requisition
+
+    StockMovementDirection getStockMovementDirection(Location currentLocation) {
+        if (currentLocation == shipment?.origin) {
```

Comment:
I'm not sure if we could get rid of the `origin` and `destination` properties and just use the shipment ones. Logically, it would be ok, but looking at the view (`stock-movement-list-item` or `stock-movement`) the `origin` and `destination` is taken from the requisition OR order (if it's return), not from the shipment. 
Probably there is no chance the shipment origin/destination pair would be different from the requisition/order's origin/destination, but maybe we should be safer?